# Ozone study

## Loading the dataset

In [6]:
library(leaps)

(ozone_df <- read.csv("ozone.csv", sep=" "))

maxO3 T9   T12  T15  Ne9 Ne12 Ne15 Vx9     Vx12    Vx15    maxO3v
20010601  87   15.6 18.5 18.4 4   4    8     0.6946 -1.7101 -0.6946  84   
20010602  82   17.0 18.4 17.7 5   5    7    -4.3301 -4.0000 -3.0000  87   
20010603  92   15.3 17.6 19.5 2   5    4     2.9544  1.8794  0.5209  82   
20010604 114   16.2 19.7 22.5 1   1    0     0.9848  0.3473 -0.1736  92   
20010605  94   17.4 20.5 20.4 8   8    7    -0.5000 -2.9544 -4.3301 114   
20010606  80   17.7 19.8 18.3 6   6    7    -5.6382 -5.0000 -6.0000  94   
20010607  79   16.8 15.6 14.9 7   8    8    -4.3301 -1.8794 -3.7588  80   
20010610  79   14.9 17.5 18.9 5   5    4     0.0000 -1.0419 -1.3892  99   
20010611 101   16.1 19.6 21.4 2   4    4    -0.7660 -1.0261 -2.2981  79   
20010612 106   18.3 21.9 22.9 5   6    8     1.2856 -2.2981 -3.9392 101   
20010613 101   17.3 19.3 20.2 7   7    3    -1.5000 -1.5000 -0.8682 106   
20010614  90   17.6 20.3 17.4 7   6    8     0.6946 -1.0419 -0.6946 101   
20010615  72   18.3 19.6 19.4 7   5    6    -0.8682 -2.7362 -6.8944  90   
20010616  70   17.1 18.2 18.0 7   7    7    -4.3301 -7.8785 -5.1962  72   
20010617  83   15.4 17.4 16.6 8   7    7    -4.3301 -2.0521 -3.0000  70   
20010618  88   15.9 19.1 21.5 6   5    4     0.5209 -2.9544 -1.0261  83   
20010620 145   21.0 24.6 26.9 0   1    1    -0.3420 -1.5321 -0.6840 121   
20010621  81   16.2 22.4 23.4 8   3    1     0.0000  0.3473 -2.5712 145   
20010622 121   19.7 24.2 26.9 2   1    0     1.5321  1.7321  2.0000  81   
20010623 146   23.6 28.6 28.4 1   1    2     1.0000 -1.9284 -1.2155 121   
20010624 121   20.4 25.2 27.7 1   0    0     0.0000 -0.5209  1.0261 146   
20010625 146   27.0 32.7 33.7 0   0    0     2.9544  6.5778  4.3301 121   
20010626 108   24.0 23.5 25.1 4   4    0    -2.5712 -3.8567 -4.6985 146   
20010627  83   19.7 22.9 24.8 7   6    6    -2.5981 -3.9392 -4.9240 108   
20010628  57   20.1 22.4 22.8 7   6    7    -5.6382 -3.8302 -4.5963  83   
20010629  81   19.6 25.1 27.2 3   4    4    -1.9284 -2.5712 -4.3301  57   
20010630  67   19.5 23.4 23.7 5   5    4    -1.5321 -3.0642 -0.8682  81   
20010701  70   18.8 22.7 24.9 5   2    1     0.6840  0.0000  1.3681  67   
20010702 106   24.1 28.4 30.1 0   0    1     2.8191  3.9392  3.4641  70   
20010703 139   26.6 30.1 31.9 0   1    4     1.8794  2.0000  1.3681 106   
⋮        ⋮     ⋮    ⋮    ⋮    ⋮   ⋮    ⋮    ⋮       ⋮       ⋮       ⋮     
20010828 112   21.0 24.4 26.8 1   6    3     4.0000  4.0000  3.7588 114   
20010829 101   16.9 17.8 20.6 7   7    7    -2.0000 -0.5209  1.8794 112   
20010830  76   17.5 18.6 18.7 7   7    7    -3.4641 -4.0000 -1.7321 101   
20010831  59   16.5 20.3 20.3 5   7    6    -4.3301 -5.3623 -4.5000  76   
20010901  78   17.7 20.2 21.5 5   5    3     0.0000  0.5209  0.0000  59   
20010902  76   17.3 22.7 24.6 4   5    6    -2.9544 -2.9544 -2.0000  78   
20010903  55   15.3 16.8 19.2 8   7    5    -1.8794 -1.8794 -2.3941  76   
20010904  71   15.9 19.2 19.5 7   5    3    -6.1284  0.0000 -1.3892  55   
20010905  66   16.2 18.9 19.3 2   5    6    -1.3681 -0.8682  1.7101  71   
20010906  59   18.3 18.3 19.0 7   7    7    -3.9392 -1.9284 -1.7101  66   
20010907  68   16.9 20.8 22.5 6   5    7    -1.5000 -3.4641 -3.0642  59   
20010908  63   17.3 19.8 19.4 7   8    8    -4.5963 -6.0622 -4.3301  68   
20010912  78   14.2 22.2 22.0 5   5    6    -0.8660 -5.0000 -5.0000  62   
20010913  74   15.8 18.7 19.1 8   7    7    -4.5963 -6.8937 -7.5175  78   
20010914  71   15.2 17.9 18.6 6   5    1    -1.0419 -1.3681 -1.0419  74   
20010915  69   17.1 17.7 17.5 6   7    8    -5.1962 -2.7362 -1.0419  71   
20010916  71   15.4 17.7 16.6 4   5    5    -3.8302  0.0000  1.3892  69   
20010917  60   13.7 14.0 15.8 4   5    4     0.0000  3.2139  0.0000  71   
20010918  42   12.7 14.3 14.9 8   7    7    -2.5000 -3.2139 -2.5000  60   
20010919  65   14.8 16.3 15.9 7   7    7    -4.3301 -6.0622 -5.1962  42   
20010920  71   15.5 18.0 17.4 7   7    6    -3.9392 -3.0642  0.0000  65   
20010921  96   11.3 19.4 20.2 3   

## Question 1: Chose a criterion to predict the ozon level

Y = Max(O3)
Explain Y with columns from 2 to 13 with another variable using a linear model.

In [7]:
interesting_columns <- c("T9", "T12", "T15", "Ne9", "Ne12", "Ne15", "Vx9", "Vx12", "Vx15", "maxO3v")

r_square <- c()

max <- 0

for(col in interesting_columns){
    r_squared <- summary(lm(ozone_df$maxO3~ozone_df[, col]))$r.squared
    r_square <- c(r_square, r_squared)
    
    if (r_squared > max){
        best_variable <- col
        max <- r_squared
    }
}

print(max)
print(best_variable)

[1] 0.6150674
[1] "T12"


We determined that with only one vector, the variable that builds the best linear model is "T12" with a r square of 0.6150674.

## Question 2: Launch automatic selections


### Forward


In [23]:
best_models_10_forward <- regsubsets(maxO3~.,
                                    ozone_df,
                                    nvmax =10,
                                    method = "forward")
a<-summary(best_models_10_forward)

out_df <- data.frame(row.names <- 1:10,
                     a$rsq, a$adjr2, a$bic, a$cp)
colnames(out_df) <- c("Nb Variables", "R2", "Adj R2", "BIC", "CP")

out_df

Nb Variables R2        Adj R2    BIC        CP        
1   1           0.6150674 0.6115680  -97.48795 53.3384272
2   2           0.7012408 0.6957590 -121.15365 19.2202055
3   3           0.7519764 0.7450869 -137.27994 -0.0448404
4   4           0.7622198 0.7533308 -137.28524 -2.3381610
5   5           0.7631187 0.7519451 -132.99099 -0.7149565
6   6           0.7642858 0.7508165 -128.82567  0.7958717
7   7           0.7658186 0.7500564 -124.83783  2.1534559
8   8           0.7665145 0.7483797 -120.45266  3.8617710
9   9           0.7671134 0.7465646 -116.02184  5.6107249
10 10           0.7674514 0.7444268 -111.46599  7.4690791

### Backward


In [24]:
best_models_10_backward <- regsubsets(maxO3~.,
                                    ozone_df,
                                    nvmax =10,
                                    method = "backward")
a<-summary(best_models_10_forward)

out_df <- data.frame(row.names <- 1:10,
                     a$rsq, a$adjr2, a$bic, a$cp)
colnames(out_df) <- c("Nb Variables", "R2", "Adj R2", "BIC", "CP")

out_df

Nb Variables R2        Adj R2    BIC        CP        
1   1           0.6150674 0.6115680  -97.48795 53.3384272
2   2           0.7012408 0.6957590 -121.15365 19.2202055
3   3           0.7519764 0.7450869 -137.27994 -0.0448404
4   4           0.7622198 0.7533308 -137.28524 -2.3381610
5   5           0.7631187 0.7519451 -132.99099 -0.7149565
6   6           0.7642858 0.7508165 -128.82567  0.7958717
7   7           0.7658186 0.7500564 -124.83783  2.1534559
8   8           0.7665145 0.7483797 -120.45266  3.8617710
9   9           0.7671134 0.7465646 -116.02184  5.6107249
10 10           0.7674514 0.7444268 -111.46599  7.4690791

### Sequential


In [10]:
best_models_10_seqrep <- regsubsets(maxO3~.,
                                    ozone_df,
                                    nvmax =10,
                                    method = "seqrep")
a<-summary(best_models_10_forward)

out_df <- data.frame(a$rsq, a$adjr2, a$bic, a$cp,row.names<-interesting_columns)
colnames(out_df) <- c("R2", "Adj R2", "BIC", "CP", "Variables")

out_df

R2        Adj R2    BIC        CP         Variables
1  0.6150674 0.6115680  -97.48795 53.3384272 T9       
2  0.7012408 0.6957590 -121.15365 19.2202055 T12      
3  0.7519764 0.7450869 -137.27994 -0.0448404 T15      
4  0.7622198 0.7533308 -137.28524 -2.3381610 Ne9      
5  0.7631187 0.7519451 -132.99099 -0.7149565 Ne12     
6  0.7642858 0.7508165 -128.82567  0.7958717 Ne15     
7  0.7658186 0.7500564 -124.83783  2.1534559 Vx9      
8  0.7665145 0.7483797 -120.45266  3.8617710 Vx12     
9  0.7671134 0.7465646 -116.02184  5.6107249 Vx15     
10 0.7674514 0.7444268 -111.46599  7.4690791 maxO3v

## Question 3: Launch automatic selections
